src/common/Configuration/Config.cpp

```diff
@@ -356,28 +356,30 @@ template<class T>
 T ConfigMgr::GetValueDefault(std::string const& name, T const& def, bool showLogs /*= true*/) const
 {
     std::string strValue;
+
     auto const& itr = _configOptions.find(name);
-    if (itr == _configOptions.end())
+    bool notFound = itr == _configOptions.end();
+    Optional<std::string> envVar = EnvVarForIniKey(name);
```

Comment:
I have a concern with this approach. In this approach, every function call to retrieve a certain config value would result in a syscall to retrieve an environment variable. And since not all config values are cached, with these changes it could potentially impact performance. Personally, I would prefer to load and cache all environment variables during startup.

Reply:
Yeah, this was in the back of my mind while I was working on this, but I was worried it'd be a premature optimization. 

It can't hurt though, I'll probably be able to make this change evening.

Reply:
@walkline Added a cache of env vars